cheat.php

```diff
@@ -713,6 +775,55 @@ function ExecuteRequest( $Method, $URL, $Data = [] )
 	return $Data;
 }
 
+function GetLocalScriptHash( )
+{
+	return sha1_file( __FILE__ );
+}
+
+function GetRepositoryScriptHash( )
+{
+	$c_r = curl_init( );
+
+	curl_setopt_array( $c_r, [
+		CURLOPT_USERAGENT      => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3464.0 Safari/537.36',
+		CURLOPT_RETURNTRANSFER => true,
+		CURLOPT_ENCODING       => 'gzip',
+		CURLOPT_TIMEOUT        => 15,
+		CURLOPT_CONNECTTIMEOUT => 5,
+		CURLOPT_HEADER         => 1,
+		CURLOPT_CAINFO         => __DIR__ . '/cacert.pem',
+	] );
+
+	curl_setopt( $c_r, CURLOPT_URL, 'https://api.github.com/repos/SteamDatabase/SalienCheat/git/trees/master' );
+	curl_setopt( $c_r, CURLOPT_HTTPGET, 1 );
+
+	do
+	{
+		$Data = curl_exec( $c_r );
+
+		$HeaderSize = curl_getinfo( $c_r, CURLINFO_HEADER_SIZE );
+		$Data = substr( $Data, $HeaderSize );
+		$Data = json_decode( $Data, true );
+
+		if ( isset( $Data[ 'tree' ] ) )
+		{
+			foreach( $Data[ 'tree' ] as &$File )
+			{
+				if ( isset( $File[ 'path' ] ) && $File[ 'path' ] === "cheat.php" )
+				{
+					if ( isset( $File[ 'sha' ]) )
+						return $File[ 'sha' ];
+
+					break;
```

Comment:
I doubt the 2 isset checks above are necessary to have.

And do `return sha` instead of this break.

Reply:
Sanity checks.

break ensures that the foreach loop is no longer called (since we found the file) and goes to the error message below.

Reply:
Yeah just remove these issets.

If you `return`, then you don't need `break`

Reply:
Are you sure? Corrupted data could cause a crash.

Reply:
In php the worst thing that will happen is a notice message.

And I doubt the github api is so bad to return a tree but not sha or path.